ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java

```diff
@@ -259,12 +278,54 @@ public void onInitializeAccessibilityNodeInfo(View host, AccessibilityNodeInfoCo
     }
   }
 
+  private boolean isViewVisible(View scrollView, View view) {
+    Rect scrollBounds = new Rect();
+    scrollView.getDrawingRect(scrollBounds);
+    float viewHeight = view.getHeight();
+    // Verify View is half visible
+    float top = view.getY() + viewHeight / 2;
+    float bottom = top + view.getHeight() - viewHeight / 2;
+
+    if (scrollBounds.top < top  && scrollBounds.bottom > bottom) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+  
   @Override
   public void onInitializeAccessibilityEvent(View host, AccessibilityEvent event) {
     super.onInitializeAccessibilityEvent(host, event);
     // Set item count and current item index on accessibility events for adjustable
     // in order to make Talkback announce the value of the adjustable
     final ReadableMap accessibilityValue = (ReadableMap) host.getTag(R.id.accessibility_value);
+    final ReadableMap accessibilityCollectionInfo = (ReadableMap) host.getTag(R.id.accessibility_collection_info);
+    if (accessibilityCollectionInfo != null) {
+      event.setItemCount(accessibilityCollectionInfo.getInt("rowCount"));
+
+      View contentView = ((ViewGroup) host).getChildAt(0);
+
+       ReadableMap firstVisible = null;
+       ReadableMap lastVisible = null;
+
+      for(int index = 0; index < ((ViewGroup) contentView).getChildCount(); index++) {
+        View nextChild = ((ViewGroup) contentView).getChildAt(index);
```

Comment:
And here.

Reply:
https://github.com/facebook/react-native/pull/31666/commits/ec09520b4086952f04876b2a1e1e07ad888e5a32#diff-7746c3776d60ee249dca2e6e101b68ef31b40c72c9a02772709e8d95f93e72d8R137-R139

Reply:
https://github.com/facebook/react-native/pull/31666/commits/ec09520b4086952f04876b2a1e1e07ad888e5a32#diff-7746c3776d60ee249dca2e6e101b68ef31b40c72c9a02772709e8d95f93e72d8R137-R139